ptf/saimpls.py

```diff
@@ -0,0 +1,3818 @@
+            verify_packet(self, recv_pkt, self.dev_port13)
+            self.ingress_rif_in_stats += 1
+            self.egress_rif_3_out_stats += 1
+
+            print("Send MPLS tag pakcet with labels - 2111, 2000 - "
+                  "PHP and forward IP packet")
+            send_packet(self, self.dev_port10, mpls_tag_2_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+            self.ingress_rif_in_stats += 1
+            self.egress_rif_3_out_stats += 1
+
+            print("Send MPLS tag pakcet with labels - 2222, 2111, 2000 - "
+                  "PHP and forward IP packet")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+            self.ingress_rif_in_stats += 1
+            self.egress_rif_3_out_stats += 1
+
+            print("Set MPLS object label 2222 to pop only 2 labels")
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=2)
+            print("Send MPLS tag pakcet with labels - 2222, 2111, 2000 - "
+                  "Pop 2 labels and forward MPLS with 2000")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_mpls_pkt, self.dev_port13)
+            self.ingress_rif_in_stats += 1
+            self.egress_rif_3_out_stats += 1
+
+            self.assertTrue(self._verifyStats())
+
+        finally:
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=3)
+
+    def mplsEgressPhpSwapNullTest(self):
+        '''
+        Verfies PHP swaps label with explicit null and forwards packet.
+        '''
+        print("\nmplsEgressPhpSwapNullTest()")
+
+        send_pkt = simple_tcpv6_packet(
+            eth_dst=ROUTER_MAC,
+            ipv6_dst='400::1',
+            ipv6_hlim=64)
+
+        mpls_tag = {'label': 3000, 'ttl': 64, 'tc':  0, 's':  1}
+        mpls_tag_list = [mpls_tag]
+        mpls_pkt = simple_mpls_packet(
+            eth_dst=ROUTER_MAC,
+            mpls_tags=mpls_tag_list,
+            inner_frame=send_pkt['IPv6'])
+
+        null_tag = {'label': 0, 'ttl': 63, 'tc':  0, 's':  1}
+        null_mpls_tag_list = [null_tag]
+        recv_mpls_pkt = simple_mpls_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            mpls_tags=null_mpls_tag_list,
+            inner_frame=send_pkt['IPv6'])
+
+        print("Send MPLS tag pakcet with label 3000 - "
+              "PHP and forward packet with explicit NULL label")
+        send_packet(self, self.dev_port10, mpls_pkt)
+        verify_packet(self, recv_mpls_pkt, self.dev_port25)
+        self.ingress_rif_in_stats += 1
+        self.egress_rif_4_out_stats += 1
+
+        self.assertTrue(self._verifyStats())
+
+    def mplsTransitSwapTest(self):
+        '''
+        Verify MPLS label is swapped with another label or explicit null label
```

Comment:
IMO we could use such imperative form in all docstrings (I mean 'verify' instead of 'verifies')